src/funcx_common/redis_endpoint_lock.py

```diff
@@ -0,0 +1,108 @@
+        redis_client: "redis.Redis[t.Any]",
+        endpoint_id: str,
+        *,
+        lock_timestamp: t.Optional[float] = None,
+    ):
+        """
+        :param Redis redis_client: Redis client for properties to get/set
+        :param str endpoint_id: UUID of the task, as str
+        :param float lock_timestamp: The epoch timestamp when the lock ends
+        """
+        # non-RedisField attributes of a RedisEndpointLock
+        self.hname = f"endpoint_lock_{endpoint_id}"
+        self.redis_client = redis_client
+        self.endpoint_id = endpoint_id
+        if lock_timestamp is not None:
+            now_ts = datetime.now().timestamp()
+            if lock_timestamp <= now_ts:
+                raise ValueError("Requested lock timestamp must be in the future")
+
+            # This is stored as a float in the data, for our usage
+            self.lock_expiration_ts = lock_timestamp
+            # expire_ts is the internal reddit expiry, millisecond precision
+            self.expire_ts = lock_timestamp
+
+    @property
+    def expire_ts(self) -> float:
+        # TODO replace with pexpiretime() on upgrade
+        # pexpiretime() should return what we need but it's apparently not
+        # in the version of redis library we use?
+        # Note that this is ms precision as pexpireat setter is ms
+        now_ts = datetime.now().timestamp()
+        return now_ts + self.redis_client.pttl(self.hname) / 1000
+
+    @expire_ts.setter
+    def expire_ts(self, expiration_ts: float) -> None:
+        """Expires lock after expiration time, if not already set."""
+        ttl_ms = self.redis_client.pttl(self.hname)
+        if ttl_ms < 0:
+            expiration_ts_ms = int(expiration_ts * 1000)
+            self.redis_client.pexpireat(self.hname, expiration_ts_ms)
+
+    def delete(self) -> None:
+        """Removes this lock from Redis, not used for initial expiring case"""
+        self.redis_client.delete(self.hname)
+
+    @classmethod
+    def exists(cls, redis_client: "redis.Redis[t.Any]", endpoint_id: str) -> bool:
+        """Check if a given lock exists"""
+        return bool(redis_client.exists(f"endpoint_lock_{endpoint_id}"))
+
+    @classmethod
+    def load(
+        cls, redis_client: "redis.Redis[t.Any]", endpoint_id: str
+    ) -> "RedisEndpointLock":
+        """
+        Returns an object lock if it exists, otherwise None
+        """
+        if cls.exists(redis_client, endpoint_id):
+            return cls(redis_client, endpoint_id)
+        raise ValueError(f"Cannot find lock {endpoint_id}")
+
+    @classmethod
+    def is_locked(cls, redis_client: "redis.Redis[t.Any]", endpoint_id: str) -> bool:
+        # We don't need the info in the stored object, just using built in
+        # Redis expiry functionality to see if there's a live lock
+        # TODO add more information to the stored object and possibly
+        # checking the stored expiry timestamp
+        return cls.exists(redis_client, endpoint_id)
```

Comment:
Seems that either `.exists()` or this method is redundant, no?  What's the compelling reason to have both?

Reply:
I used is_locked as a wrapper because the wording is more clear.  But didn't remove the exists() because it's a more generic method that all redis classes probably should have.  It just happens that exists() and is_locked() is the same for this class, but maybe in the future we have existing lock objects that are expired or in some other state?

But as they are the same as of now, removed is_locked()